libvips/iofuncs/image.c

```diff
@@ -1658,18 +1658,16 @@ vips_image_set_kill( VipsImage *image, gboolean kill )
 	image->kill = kill;
 }
 
-/* Make a name for a filename-less image. Use immediately, don't free the
- * result.
+/* Fills the given buffer with a temporary filename.
+ * Assuming that "int" might be 64 Bit wide a buffer size of 26 suffices.
  */
-static const char *
-vips_image_temp_name( void )
+void
+vips_image_temp_name( char *name, int size )
 {
-	static int serial = 0;
-	static char name[256];
-
-	vips_snprintf( name, 256, "temp-%d", serial++ );
+	static int global_serial = 0;
 
-	return( name );
+	int serial = g_atomic_int_add( &global_serial, 1);
```

Comment:
It looks like [g_atomic_int_add](https://developer.gnome.org/glib/stable/glib-Atomic-Operations.html#g-atomic-int-add) only started returning a value from glib v2.30+ so we might need to check this is available at install time. I think RHEL 6 (EOL 2020) ships with an older glib version, perhaps v2.28.

Reply:
Oops, sorry, I misread your patch, you've moved the static buffer to the caller, sorry.

OK, this sounds right. As Lovell says, we'll probably need to add to configure check and switch to `g_atomic_int_exchange_and_add()` before glib 2.30.

Reply:
@jcupitt @lovell I pushed ab6126d. Please let me know if that suffices. 🙂